tokens/figma/update-skins.mjs

```diff
@@ -0,0 +1,202 @@
+import {
+  updateCollections,
+  updateOrCreateVariables,
+  updateOrCreateVariableModeValues,
+} from "./utils/figma-utils.mjs";
+
+import {
+  VARIABLE_TYPES,
+  COLLECTION_NAMES,
+  MODE_NAMES,
+} from "./utils/constants.mjs";
+
+import {
+  getFigmaData,
+  postFigmaVariables,
+} from "./utils/api-request.mjs";
+
+import { getPaletteVariables } from "./variables.mjs";
+
+const collectionNames = [
+  COLLECTION_NAMES.PALETTE,
+];
+
+async function updatePalette(
+  jsonData,
+  brand,
+  FILE_KEY,
+  FIGMA_TOKEN
+) {
+  try {
+    const figmaData = await getFigmaData(
+      FILE_KEY,
+      FIGMA_TOKEN
+    );
+
+    const existingVariables =
+      figmaData.meta.variables;
+    const existingCollections =
+      figmaData.meta.variableCollections;
+
+    const newData = {
+      variableCollections: [],
+      variableModes: [],
+      variables: [],
+      variableModeValues: [],
+    };
+
+    const paletteVariables = getPaletteVariables(
+      jsonData,
+      brand
+    );
+
+    for (const group of paletteVariables) {
+      const {
+        variables,
+        collectionName,
+        resolvedType,
+        variableScopes,
+        hasAlias,
+      } = group;
+
+      for (const variable of variables) {
+        // Update or create the variable in the collection
+        const variablesUpdateResult =
+          await updateOrCreateVariables({
+            variable: {
+              ...variable,
+              resolvedType: resolvedType,
+              scopes: variableScopes,
+              hasAlias: hasAlias,
+            },
+            targetCollectionName: collectionName,
+            existingVariables: existingVariables,
+            existingCollections:
+              existingCollections,
+          });
+
+        if (!newData.variables) {
```

Comment:
can this happen? You already initialize it as an empty array

Reply:
True is already initialized in line 44, removed